### PR TITLE
feat: metric card component

### PR DESCRIPTION
**Pull request by itssharmasandeep**
## Description
Metric card component

### Testing
Local testing done.

### Checklist:
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: projects/components/src/metric-card/metric-card.component.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+                [icon]="this.icon"
+                size="${IconSize.Small}"
+                [color]="this.indicatorColor"
+              ></ht-icon>
+            </ng-container>
+          </ng-container>
+        </div>
+        <div class="title-text">{{ this.titleText }}</div>
+      </div>
+      <div class="value">{{ value }}</div>
```
Comment (arjunlalb) [inline projects/components/src/metric-card/metric-card.component.ts:30]: we should apply the htDisplayNumber pipe if this is a number and htDisplayString pipe if its a string.

File: projects/components/src/metric-card/metric-card.component.ts [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    </div>
+  `
+})
+export class MetricCardComponent {
+  @Input()
+  public titleText?: string;
```
Comment (arjunlalb) [inline projects/components/src/metric-card/metric-card.component.ts:36]: can be simplified to `title`

File: projects/components/src/metric-card/metric-card.component.ts [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+  public indicator: MetricCardIndicatorType = MetricCardIndicatorType.Dot;
+
+  @Input()
+  public icon?: string; // Only be used when indicator type is Icon
+
+  @Input()
+  public indicatorColor: Color = Color.Gray7;
+
+  @Input()
+  public backgroundColor: Color = Color.OffWhite;
```
Comment (arjunlalb) [inline projects/components/src/metric-card/metric-card.component.ts:51]: this should also default to white/transparent.

File: projects/components/src/metric-card/metric-card.ts [hunk 4]
```diff
@@ -0,0 +1,4 @@
+export const enum MetricCardIndicatorType {
+  Dot = 'dot',
+  Icon = 'icon'
```
Comment (arjunlalb) [inline projects/components/src/metric-card/metric-card.ts:3]: `None` also should be supported if we just want a title and value.

**Review by arjunlalb — APPROVED** [review]
Comments can be addressed in a follow up.

File: projects/components/src/metric-card/metric-card.component.ts (hunk 2 above)
Comment (itssharmasandeep) [reply projects/components/src/metric-card/metric-card.component.ts:36]: Actually, I do not want that, this basically collide with HTML `title` property and we can get weird tooltip

**Pull request merged**